src/validate.js

```diff
@@ -2429,30 +2429,31 @@ function F_VALIDATE(disabledHL) {
 
 			// check Public connection
 			if (slowChecks
-				&& segment.$isRoutable){
+				&& segment.$isRoutable
+				&& nodeA.$otherSegments.length > 0
```

Comment:
`$otherSegments` has a getter -- please see `getOtherSegments()`. Using `$otherSegments.length` will trigger the getter and hence will call segment constructors for all the segments connected to nodeA.

We better use `$otherSegmentsLen` to postpone (or maybe avoid) those calls.